src/client/index.ts

```diff
@@ -153,21 +152,6 @@ function getCollectKeyFromCommand(command: string): string | null {
   }
 }
 
-/**
- * It returns a function that prepends params to the given func.
- * A sugar function for JavaScript .bind() without the "this" (keyword) binding.
- *
- * @param func - A function to prepend params.
- * @param params - Parameters to prepend to a function.
- * @returns A function bound with params.
- */
-// TODO Need to refactor prepend so TS can infer the correct function signature type
```

Comment:
We get rid of this TODO by getting rid of `prepend`

Reply:
Why are we removing prepend? (Does it mess up typescript?)

If I understand correctly the prepend's purpose here is to turn:
`client.autofill(client, transaction)` into `client.autofill(transaction)` which seems like a nice simplification from the interface perspective.

Reply:
Oh, are these functions now like instance methods, just defined outside the class?

Reply:
pretty much yeah

Reply:
Before deleting this, I'd like to see if there's a possible fix to see if we can have TS infer the correct result function signature type. I think there's value keeping `prepend`.

Reply:
So, there is probably a way to get this to work, but I don't personally see the benefit of explicitly binding `this` to a member of `Client`. Especially when typescript will do it implicitly for us by naming the `Client` parameter `this`.